packages/core/src/SpringRef.ts

```diff
@@ -7,9 +7,28 @@ interface ControllerUpdateFn<State extends Lookup = Lookup> {
   (i: number, ctrl: Controller<State>): ControllerUpdate<State> | Falsy
 }
 
-export class SpringRef<State extends Lookup = Lookup> {
+/**
+ * Extending from function allows SpringRef instances to be callable.
+ * https://hackernoon.com/creating-callable-objects-in-javascript-d21l3te1
+ *
+ * ```js
+ * const [springs, animate] = useSpring(() => ({}))
+ * animate.set({}) // this works
+ * animate({}) // this also works
+ * ```
+ */
+export class SpringRef<State extends Lookup = Lookup> extends Function {
   readonly current: Controller<State>[] = []
 
+  constructor() {
+    super('return arguments.callee._call.apply(arguments.callee, arguments)')
+  }
+
+  _call(props?: ControllerUpdate<State> | ControllerUpdateFn<State>) {
+    // TODO maybe we should deprecate calling the ref directly.
```

Comment:
I would really like to push forward with enforcing the mental model of the `api` object. So for me depreciating is the right thing to do, there's a function that handles depreciation of `interpolate`.

Reply:
Do you want me to handle deprecation in this PR? I'm just a bit worried that we would throwing a LOT of warnings (one per frame 😅)

Reply:
I’ll look at it and add it to the PR and see what magic I can do, I think I have a logonce function lying around 👀